### ISSUE TITLE
when build parallel version ,meet  OSError: [WinError 123]………………msmpi.a

### ISSUE THREAD

**Issue by Zcaic** [su2code/SU2#1471]
**Describe the bug**
*A clear and concise description of what the bug is and what you expect the behavior to be instead. If applicable, add screenshots to help explain your problem.*
When I build a parallel version of SU2 under Win10, I get an error like

“OSError: [WinError 123] 文件名、目录名或卷标语法不正确。: 'D:\\mingw-w64\\x86_64\\mingw64\\bin\\..\\lib\\gcc\\x86_64-w64-mingw32\\8.1.0\\libC:\\Program Files (x86)\\Microsoft SDKs\\MPI\\Lib\\x64\\msmpi.a'”

I noticed that other people on Github have also raised this issue, but it hasn't been resolved, and I'm curious to know, what should I deal with it.
thanks！

**To Reproduce**
*Please provide a config file and mesh for a minimal example that recreates the problem.*
[wrong _information.txt](https://github.com/su2code/SU2/files/7734843/wrong._information.txt)
[meson-info_json.txt](https://github.com/su2code/SU2/files/7734868/meson-info_json.txt)

**Desktop (please complete the following information):**
 - OS: [win10 ]
 - C++ compiler and version: [ g++ (GCC) 8.1.0]
 - MPI implementation and version: [Microsoft MPI v10.1.2]
 - SU2 Version: [v7.2.1]

**Comment by pcarruscag** [comment]
Seems related with #1468, let's keep only one issue.

**Comment by Zcaic** [comment]
thanks，I have solved this problem, which I think is caused by the difference in the way Linux and Windows express paths

**Comment by pcarruscag** [comment]
Can you share the solution?

**Comment by Zcaic** [comment]
1) after you install msmpi, you should modify environment Variable of MSMPI_LIB64, you can type **set MSMPI_LIB64=\%MSMPI_LIB64%** in cmd or **$env:MSMPI_LIB64="\\"+$env:MSMPI_LIB64** in powershell, then you can build code like "python meson.py build -Denable-tecio=false -Denable-autodiff=true --prefix=D:\SU2\executable", it will work this time;
2)after Configuration, you should find the file named "build.ninja" in file called "build", then open it as a text editor, you can look for string "**-l\\\\**" through the entire file, you wiil find several of strings like "-l\\\\D:\\\\mpi\\\\Microsoft_SDKs\\\\MPI\\\\Lib\\\\x64\\\\msmpi", it's the path of msmpi installation, it may be a little different in you computer, then you should replace all strings with "D:\\\\mpi\\\\Microsoft_SDKs\\\\MPI\\\\Lib\\\\x64\\\\msmpi.dll", it's path of **msmpi.dll**, please use your own path;
3) Last ,type **ninja -C build install** to compile，this time it will be all right.

**Comment by pcarruscag** [comment]
Thank you, that looks like the build system is doing something weird, unfortunately I don't have a machine with windows to test.

**Comment by zhaokuohaha** [comment]
> 

Hi zcaic, 

I tried this,  but it still thow the same error.  with message
```
OSError: [WinError 123] 文件名、目录名或卷标语法不正确。: 'D:\\mingw-w64\\x86_64\\mingw64\\bin\\..\\lib\\gcc\\x86_64-w64-mingw32\\8.1.0\\lib\\C:\\Program Files (x86)\\Microsoft SDKs\\MPI\\Lib\\x64\\msmpi.a'
```

⬆️  just from `xxx\\8.1.0\\libC:\\Program...` to `xxx\\8.1.0\\libC:\\Program...`

Is that any solutions ?

**Comment by Zcaic** [comment]
@zhaokuohaha , did you add "/" before "%MSMPI_LIB64%"? just like "set MSMPI_LIB64=\%MSMPI_LIB64%" in cmd

**Comment by zhaokuohaha** [comment]
@Zcaic  nope, I add '\\' before`'C:Program Files...`  and the environment MSMPI_LIB64  finally like `\C:\Program Files (x86)\Microsoft SDKs\MPI\Lib\x64\`

 I paste wrong error message before . the actual error message after I change the envirnment variable is 
 from `xxx\\8.1.0\\libC:\\Program...` to `xxx\\8.1.0\\lib\\C:\\Program...`  .   just a more `\\`

**Comment by zhaokuohaha** [comment]
OK I Solved it  by use  version  7.2.    this problem still exists for version 7.3.1


> @Zcaic nope, I add '\' before`'C:Program Files...` and the environment MSMPI_LIB64 finally like `\C:\Program Files (x86)\Microsoft SDKs\MPI\Lib\x64\`
> 
> I paste wrong error message before . the actual error message after I change the envirnment variable is from `xxx\\8.1.0\\libC:\\Program...` to `xxx\\8.1.0\\lib\\C:\\Program...` . just a more `\\`

**Comment by Zcaic** [comment]
When compiling the V7.4 parallel version of SU2 on win10, you should change the msmpi link path in the build/build.ninja file, Such as "- lD:\MPI\SDKs\MPI\Lib\x64\msmpi" to "D:\\MPI\\SDKs\\MPI\\Lib\\x64\\msmpi.Lib"
